### PR TITLE
Autocomplete

### PR DESCRIPTION
**Pull request by ioanam25**
https://drive.google.com/file/d/1Qqm2kyqPYZWVF-X3MnkX-GodnEA5nnCv/view?usp=sharing

implemented autocomplete suggestions for name and category

https://drive.google.com/file/d/1A0nxgXDBY5ACWXiNZ09wUigs5A10uG9v/view?usp=sharing
suggestions for keywords

### CODE REVIEW

**Review by Zelin-Zeng — APPROVED** [review]

File: app/src/main/java/com/example/meetup/Fragments/SearchFragment.java [hunk 1]
```diff
@@ -0,0 +1,391 @@
...
+
+    // A  Thread on which the Api request will be made and results will be delivered. As network calls cannot be made on the amin thread, so we are creating a separate thread for the network calls
+    private Thread thread;
+
+    // Google Request for the NLP Api. This actually is acting like a Http client queue that will process each request and response from the Google Cloud server
+    private final BlockingQueue<CloudNaturalLanguageRequest<? extends GenericJson>> requests
+            = new ArrayBlockingQueue<>(100);
+
+    // Api for CloudNaturalLanguage from the Google Client library, this is the instance of our request that we will make to analyze the text.
+    private CloudNaturalLanguage api = new CloudNaturalLanguage.Builder(
```
Comment (Zelin-Zeng) [inline app/src/main/java/com/example/meetup/Fragments/SearchFragment.java:88]: Well, it's better to move the initialization to constructor

File: app/src/main/java/com/example/meetup/Fragments/SearchFragment.java [hunk 2]
```diff
@@ -0,0 +1,391 @@
...
+                    credential.initialize(request);
+                }
+            }).build();
+
+
+    public SearchFragment() {
+        // Required empty public constructor
+    }
+
+    interface Function {
```
Comment (Zelin-Zeng) [inline app/src/main/java/com/example/meetup/Fragments/SearchFragment.java:103]: Rename it to QueryResponseCallback
And onCalled should be renamed to onPostsReturned

File: app/src/main/java/com/example/meetup/Fragments/SearchFragment.java [hunk 3]
```diff
@@ -0,0 +1,391 @@
...
+                    String textDescription = post.getDescription();
+                    analyzeTextUsingCloudNLPApi(textDescription);
+
+                    String textCaption = post.getCaption();
+                    analyzeTextUsingCloudNLPApi(textCaption);
+
+                }
+                searchNameAdapter.notifyDataSetChanged();
+                searchCategoriesAdapter.notifyDataSetChanged();
+                searchRolesAdapter.notifyDataSetChanged();
```
Comment (Zelin-Zeng) [inline app/src/main/java/com/example/meetup/Fragments/SearchFragment.java:187]: Do we need to notify adapter here?

File: app/src/main/java/com/example/meetup/Fragments/SearchFragment.java [hunk 4]
```diff
@@ -0,0 +1,391 @@
...
+                }
+
+                // save received posts to list and notify adapter of new data
+                allPosts.addAll(posts);
+                adapter.notifyDataSetChanged();
+            }
+        });
+    }
+
+    private void getColumnArray(Function callback) {
```
Comment (Zelin-Zeng) [inline app/src/main/java/com/example/meetup/Fragments/SearchFragment.java:248]: Suggest: there seems duplicate part between getColumnArray and queryPosts. I think we can just keep queryPosts with callback.

File: app/src/main/java/com/example/meetup/Fragments/SearchFragment.java (hunk 1 above)
Comment (ioanam25) [reply app/src/main/java/com/example/meetup/Fragments/SearchFragment.java:88]: isn't the constructor required to be empty for a fragment?

File: app/src/main/java/com/example/meetup/Fragments/SearchFragment.java (hunk 4 above)
Comment (ioanam25) [reply app/src/main/java/com/example/meetup/Fragments/SearchFragment.java:248]: the query is different, in query posts i only want select posts while in getColumnArray i want all the entries. I'm not sure how to get rid of the duplicate code

**Pull request merged**